packages/flutter/lib/src/widgets/text.dart

```diff
@@ -703,8 +714,19 @@ class Text extends StatelessWidget {
     if (style == null || style!.inherit) {
       effectiveTextStyle = defaultTextStyle.style.merge(style);
     }
-    if (MediaQuery.boldTextOf(context)) {
-      effectiveTextStyle = effectiveTextStyle!.merge(const TextStyle(fontWeight: FontWeight.bold));
+    final bool boldText = MediaQuery.boldTextOf(context);
+    final double? lineHeightScaleFactor = MediaQuery.maybeLineHeightScaleFactorOverrideOf(context);
+    final double? letterSpacing = MediaQuery.maybeLetterSpacingOverrideOf(context);
+    final double? wordSpacing = MediaQuery.maybeWordSpacingOverrideOf(context);
+    if (boldText || lineHeightScaleFactor != null || letterSpacing != null || wordSpacing != null) {
+      effectiveTextStyle = effectiveTextStyle!.merge(
+        TextStyle(
+          height: lineHeightScaleFactor,
```

Comment:
This is just setting the base style of the paragraph, so it does not guarantee all text to have the same `height`, `letterSpacing` or `wordSpacing`, a TextSpan can still override the style (`bold` behaves the same way, but `textScaler` goes into spans). What's the desired behavior here? If the TextSpan is dynamically converted from markdown I think in a lot of cases the text won't respect the overrides.

Reply:
Yeah I think it's more reasonable to have this be automatic for all text spans, I think we can do something similar to text scaling where we plumb it down to `TextPainter` and add some additional fields to `TextStyle.getStyle` and `InlineSpan.build`.

Reply:
> > The alternative is that the developer will have to do that?
> 
> That's correct, but doesn't that mirror other text system settings like bolding or size scaling? Do you have an alternative implementation in mind?

`boldText` only applies to the paragraph base style, but text scaler applies to every `InlineSpan` in the `InlineSpan` tree.

Reply:
Yeah that would be a breaking change for implementers of `InlineSpan` and `TextPainter`. A private `InlineSpan` subclass that overrides its style sounds reasonable to me, we would need to use it in both `EditableText` and `Text` and wrap whatever span the user provides.